wa/workloads/pcmark/__init__.py

```diff
@@ -17,55 +17,76 @@
+              re.compile(r'PcmaDataManipulationScore>([\d.]+)'),
+              re.compile(r'PcmaPhotoEditingV2Score>([\d.]+)'),
+              re.compile(r'PcmaWorkv2Score>([\d.]+)'),
+              re.compile(r'PcmaWritingV2Score>([\d.]+)')],
+
+        '3': [re.compile(r'PcmaWebV3Score>([\d.]+)'),
+              re.compile(r'PcmaVideoEditingV3Score>([\d.]+)'),
+              re.compile(r'PcmaDataManipulationV3Score>([\d.]+)'),
+              re.compile(r'PcmaPhotoEditingV3Score>([\d.]+)'),
+              re.compile(r'PcmaWorkv3Score>([\d.]+)'),
+              re.compile(r'PcmaWritingV3Score>([\d.]+)')]
+    }
+
     description = '''
-    A workload to execute the Work 2.0 benchmarks within PCMark - https://www.futuremark.com/benchmarks/pcmark-android
+    A workload to execute the Work x.0 benchmarks within PCMark - https://www.futuremark.com/benchmarks/pcmark-android
 
     Test description:
     1. Open PCMark application
     2. Swipe right to the Benchmarks screen
-    3. Select the Work 2.0 benchmark
-    4. If needed, install the Work 2.0 benchmark (requires an internet connection)
-    5. Execute the Work 2.0 benchmark
-
-    Known working APK version: 2.0.3716
+    3. Select the Work x.0 benchmark
+    4. If needed, install the Work x.0 benchmark (requires an internet connection)
+    5. Execute the Work x.0 benchmark
     '''
 
-    # Do not delete Work 2.0 data-set before each run
+    # Do not delete Work x.0 data-set before each run
     clear_data_on_reset = False
 
+    parameters = [
+        Parameter('version', allowed_values=supported_versions,
+                  description='Specifies which version of the workload should be run.',
+                  override=True)
+    ]
+
     def __init__(self, target, **kwargs):
         super(PcMark, self).__init__(target, **kwargs)
         self.gui.timeout = 1500
 
     def extract_results(self, context):
-        results_path = self.target.path.join(self.target.external_storage, "PCMark for Android")
-        result_file = self.target.list_directory(results_path)[-1]
+        if self.version.startswith('3'):
+            results_path = self.target.path.join(self.target.package_data_directory, self.package, 'files')
+            result_file = [f for f in self.target.list_directory(results_path) if f.endswith(".zip")][-1]
+        elif self.version.startswith('2'):
+            results_path = self.target.path.join(self.target.external_storage, "PCMark for Android")
+            result_file = self.target.list_directory(results_path)[-1]
+
         self.result_file = result_file.rstrip()
         result = self.target.path.join(results_path, result_file)
         self.target.pull(result, context.output_directory)
         context.add_artifact('pcmark-result', self.result_file, kind='raw')
 
     def update_output(self, context):
-        expected_results = len(self.regex_matches)
+        expected_results = len(self.regex_matches[self.version.strip()[0]])
```

Comment:
If you're reusing `self.version.strip()[0]` multiple times, it might be cleaner to have a `major_version = self.version.strip()[0]` and reuse the variable instead.